src/Kestrel.Core/Internal/Http2/Http2OutputProducer.cs

```diff
@@ -37,11 +39,13 @@ public class Http2OutputProducer : IHttpOutputProducer, IHttpOutputAborter
             Http2FrameWriter frameWriter,
             StreamOutputFlowControl flowControl,
             ITimeoutControl timeoutControl,
-            MemoryPool<byte> pool)
+            MemoryPool<byte> pool,
+            Func<HttpResponseTrailers> getTrailers)
```

Comment:
This is a bit weird and allocatey

Reply:
Yeah, it was this or passing in the entire Http2Stream which this type seemed to be avoiding. I wasn't passing in the trailers directly since they're lazily allocated.